build.gradle.kts

```diff
@@ -0,0 +1,362 @@
+        )
+    }
+}
+
+tasks.register<Task>("generate") {
+    group = "source-generation"
+    description = "Generate tag-handling code using tags description."
+
+    doLast {
+        kotlinx.html.generate.generate(
+                "kotlinx.html",
+                "src/commonMain/kotlin/generated",
+                "src/jsMain/kotlin/generated"
+        )
+    }
+}
+
+tasks.register<Copy>("jsPackagePrepare") {
+    dependsOn("jsMainClasses")
+    tasks["assemble"].dependsOn(this)
+
+    group = "build"
+    description = "Assembles NPM package (result is placed into 'build/tmp/jsPackage')."
+
+    val baseTargetDir = "$buildDir/tmp/jsPackage"
+
+    from("README-JS.md")
+    from("$buildDir/js/packages/${project.name}/kotlin")
+    into(baseTargetDir)
+
+    rename("README-JS.md", "README.md")
+    rename("kotlinx-html.js", "kotlinx-html-js.js")
+    rename("kotlinx-html.js.map", "kotlinx-html-js.js.map")
+    rename("kotlinx-html.meta.js", "kotlinx-html-js.meta.js")
+
+    doLast {
+        var npmVersion = version as String
+        if (npmVersion.endsWith("-SNAPSHOT")) {
+            npmVersion = npmVersion.replace("-SNAPSHOT", "-${System.currentTimeMillis()}")
+        }
+
+        val organization = when {
+            project.hasProperty("branch-build") -> "kotlinx-branch-build"
+            project.hasProperty("master-build") -> "kotlinx-master-build"
+            else -> null
+        }
+
+        File(baseTargetDir, "package.json").writeText(packageJson(npmVersion, organization))
+        file("$baseTargetDir/kotlinx-html").renameTo(File("$buildDir/js-module/kotlinx-html-js"))
+    }
+}
+
+tasks.register<Exec>("publishNpm") {
+    dependsOn("jsPackagePrepare")
+    dependsOn("kotlinNodeJsSetup")
+
+    group = "publishing"
+    description = "Publishes ${project.name} NPM module to 'registry.npmjs.org'."
+
+    val kotlinNodeJsSetupTask = tasks["kotlinNodeJsSetup"] as NodeJsSetupTask
+
+    // For some unknown reason, the node distributive's structure is different on Windows and UNIX.
+    val node = if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+        kotlinNodeJsSetupTask.destination
+                .resolve("node.exe")
+    } else {
+        kotlinNodeJsSetupTask.destination
+                .resolve("bin")
+                .resolve("node")
+    }
+
+    val npm = if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+        kotlinNodeJsSetupTask.destination
+                .resolve("node_modules")
+                .resolve("npm")
+                .resolve("bin")
+                .resolve("npm-cli.js")
+    } else {
+        kotlinNodeJsSetupTask.destination
+                .resolve("lib")
+                .resolve("node_modules")
+                .resolve("npm")
+                .resolve("bin")
+                .resolve("npm-cli.js")
+    }
+
+    commandLine(
+            node,
+            npm,
+            "publish",
+            "$buildDir/tmp/jsPackage",
+            "--//registry.npmjs.org/:_authToken=${System.getenv("NPMJS_AUTH")}",
```

Comment:
So any user may see the token via UNIX `ps` command. Did we do the same with Ant?

Reply:
Here is a constructor of the identical command in Ant: https://github.com/Kotlin/kotlinx.html/blob/master/npm_publish.xml#L127. I think the behavior is the same.

I don't think we can avoid such vulnerability if we use the same user or superuser. The same issue for file of npm config or anything.